pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -598,7 +598,15 @@ func ExtractEnvoyEndpoints(locEps []*LocalityEndpoints) []*endpoint.LocalityLbEn
 
 // buildEnvoyLbEndpoint packs the endpoint based on istio info.
 func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnabled bool) *endpoint.LbEndpoint {
-	addr := util.BuildAddress(e.Address, e.EndpointPort)
+	addr := util.BuildAddress(e.Addresses[0], e.EndpointPort)
+	var additionalAddrs []*endpoint.Endpoint_AdditionalAddress
```

Comment:
please add comments explaining the behavior this gives, and why we want it.

Reply:
Sure, add description and reference [additionaladdress](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/endpoint/v3/endpoint_components.proto#config-endpoint-v3-endpoint-additionaladdress)of envoy

Reply:
This is documenting _what_ we are doing. We need to explain the resulting behavior. Does envoy load balance between the two? Pick one based on some semantics? something else?

Linking to envoy docs doesn't help, as it is equally unhelpful.

Reply:
Hi @howardjohn,  okay, I add more details description for multiple addresses support endpoint of Envoy, including the original proposal.